### PR TITLE
Fix scientific format for 0

### LINKED ISSUE #10675: [BUG] ScientificFormatter shows 0 as 0.000e+00
#### Software version info
bokeh 2.1.1
python 3.7
chrome  78.0.3904.97 (Official Build) (64-bit)
macOS 10.14.6 Mojave

#### Description of expected behavior and the observed behavior
Context: 
A table with a column of floats including 0, formatted with ScientificFormatter, precision set to 3, other parameters defaulted.

Expect: 
That 0 would be rendered as 0.

Observed:
Rendered as 0.000e+00 per configured precision (default precision gives 0.0000000000e+00).

#### Complete, minimal, self-contained example code that reproduces the issue
```
from bokeh.models.widgets import DataTable, ScientificFormatter, TableColumn
from bokeh.models import ColumnDataSource
from bokeh.plotting import show

table = DataTable(
    source=ColumnDataSource(dict(x=[1e-8, 1e-4, 1e-2, 1, 100, 10000, 1000000, 0])),
    columns=[
        TableColumn(
            field='x',
            title='X',
            formatter=ScientificFormatter(precision=3),
        ),
    ],
    editable=True,
)

show(table)
```

#### Screenshots

Screenshot 2020-11-13 at 14 07 25

### PR DESCRIPTION
**Pull request by kir0ul**
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [X] issues: fixes #10675 
- [x] tests added / passed

### CODE REVIEW

File: bokehjs/test/unit/models/widgets/tables/cell_formatters.ts [hunk 1]
```diff
@@ -122,6 +122,23 @@ describe("cell_formatters module", () => {
...
+        const df = new ScientificFormatter({precision: 10})
+        expect(df.doFormat(0, 0, 10, {}, {})).to.be.equal('<div class="bk" style="text-align: left;">10</div>')
+      })
+
+      it("should apply to_fixed if value is 1e-8", () => {
+        const df = new ScientificFormatter({precision: 1e-8})
+        expect(df.doFormat(0, 0, 1e-8, {}, {})).to.be.equal('<div class="bk" style="text-align: left;">1.0e-8</div>')
+      })
+
+
```
Comment (mattpap) [inline bokehjs/test/unit/models/widgets/tables/cell_formatters.ts:141]: @kir0ul, can you remove those empty lines for the sake of consistency with the rest of bokehjs?

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks @kir0ul !!

File: bokehjs/test/unit/models/widgets/tables/cell_formatters.ts (hunk 1 above)
Comment (kir0ul) [reply bokehjs/test/unit/models/widgets/tables/cell_formatters.ts:141]: I fixed it in my fork but I didn't see it had already been merged :grimacing: 

Do you use any automatic formatter for this kind of things, like [Prettier](https://prettier.io/) for example?

Comment (bryevdv) [reply bokehjs/test/unit/models/widgets/tables/cell_formatters.ts:141]: We are still weighing options for automatic code formatting as a team. Sorry I missed this before merging!

**Comment by kir0ul** [comment]
I guess the formatting is not that important but maybe you want to cherry-pick this commit: https://github.com/kir0ul/bokeh/commit/373c504f68eebd21c5ca886927bfcabf53ab87c2, which describes the right behavior for the test?

**Comment by bryevdv** [comment]
@kir0ul if you could submit a follow up PR that would be a little easier on my end but not worries if not (just let me know)

**Comment by kir0ul** [comment]
Sure, I can do that!

**Pull request merged**